compiler/tools/docgen.nim

```
@@ -253,27 +253,35 @@ template declareClosures =
-            of meMarkdownIllformedTable:  rbackRstMarkdownIllformedTable
-            of meNewSectionExpected:      rbackRstNewSectionExpected
-            of meGeneralParseError:       rbackRstGeneralParseError
-            of meInvalidDirective:        rbackRstInvalidDirective
-            of meInvalidField:            rbackRstInvalidField
-            of meFootnoteMismatch:        rbackRstFootnoteMismatch
-            of mwRedefinitionOfLabel:     rbackRstRedefinitionOfLabel
-            of mwUnknownSubstitution:     rbackRstUnknownSubstitution
-            of mwBrokenLink:              rbackRstBrokenLink
-            of mwUnsupportedLanguage:     rbackRstUnsupportedLanguage
-            of mwUnsupportedField:        rbackRstUnsupportedField
-            of mwRstStyle:                rbackRstRstStyle
-      ))
+    let kind =
+      case msgKind:
+      of meCannotOpenFile:          rbackRstCannotOpenFile
+      of meExpected:                rbackRstExpected
+      of meGridTableNotImplemented: rbackRstGridTableNotImplemented
+      of meMarkdownIllformedTable:  rbackRstMarkdownIllformedTable
+      of meNewSectionExpected:      rbackRstNewSectionExpected
+      of meGeneralParseError:       rbackRstGeneralParseError
+      of meInvalidDirective:        rbackRstInvalidDirective
+      of meInvalidField:            rbackRstInvalidField
+      of meFootnoteMismatch:        rbackRstFootnoteMismatch
+      of mwRedefinitionOfLabel:     rbackRstRedefinitionOfLabel
+      of mwUnknownSubstitution:     rbackRstUnknownSubstitution
+      of mwBrokenLink:              rbackRstBrokenLink
+      of mwUnsupportedLanguage:     rbackRstUnsupportedLanguage
+      of mwUnsupportedField:        rbackRstUnsupportedField
+      of mwRstStyle:                rbackRstRstStyle
+    if kind in rbackErrorKinds:
+      # TODO: errors/events that are locally fatal have to handle aborting
+      #       where the event is emitted, not here via `globalReport`
```

Testing my understanding: this means that various parts of docgen should be updated to use locally fatal error reporting, rather than relying on this catchall, correct?

---

Yea, that's what I meant. Either there needs to be a dedicated "fatal error" noreturn handler, or the respective callsites need to raise an exception themselves.

The handler implementation here should just send the diagnostic to the reporting/display management, without also doing control flow. If it does do control flow, that's totally fine, but it should not be *required* to.